### PR TITLE
Kenny/isoform

### PR DESCRIPTION
**Pull request by kennyworkman**
Introduces:
    * splice aware mapping (STAR)
    * exon-exon junction extraction (regtools)
    * splice graph clustering (leafcutter)

Interactive report for interacting with splice graphs still tbd

### CODE REVIEW

**Comment by kennyworkman** [comment]
would like to pull out trimgalore as upstream of both salmon and STAR so they can be run in parallel if scientist wants both counts and splice junctions

File: wf/__init__.py [hunk 1]
```diff
@@ -409,8 +412,23 @@ def _build_index(gentrome: Path) -> Path:
...
+
+    REMOTE_PATH = f"latch:///{input.base_remote_output_dir}{input.run_name}/Quantification (salmon)/{input.sample_name}/"
+    """Remote path prefix for LatchFiles + LatchDirs"""
+
+    #
+    # Trimgalore (separate!)
+    try:
+        outputs = [do_trimgalore(input, i, x) for i, x in enumerate(input.replicates)]
+    except TrimgaloreError as e:
+        print(f"Handling failure in trmming {input.sample_name} gracefully.")
```
Comment (ayushkamat) [inline wf/__init__.py]:
```suggestion
        print(f"Handling failure in trimming {input.sample_name} gracefully.")
```

File: wf/__init__.py [hunk 2]
```diff
@@ -489,63 +493,148 @@ def remote(suffix: Optional[str] = None) -> str:
...
+
+    make_juncs = True
+    junc_path = Path(f"/root/{input.sample_name}.bam.junc")
+    if make_juncs:
+        try:
+            # TODO - gaw so bad
+            print("\tDownloading STAR map.")
+            run(
+                [
+                    "aws",
```
Comment (ayushkamat) [inline wf/__init__.py:557]: would it be better to use s5cmd here

File: wf/__init__.py [hunk 3]
```diff
@@ -565,28 +654,32 @@ def parse_salmon_warning(alert_message: str, input: TrimgaloreSalmonInput) -> st
...
             f"Unable to produce gene mapping from tximport. Error surfaced from tximport -> {e}"
         )
-        raise RuntimeError(f"Tximport error: {e}")
+        print(f"Tximport error: {e}")
+        return None
 
     for unwanted in ("logs", "cmd_info.json"):
-        path = Path(local(unwanted))
+        path = Path(f"{SALMON_DIR}/{unwanted}")
         if "." in unwanted:
```
Comment (ayushkamat) [inline wf/__init__.py]: this is hacky, use `path.exists()` and `path.is_file()` instead

File: wf/__init__.py [hunk 4]
```diff
@@ -867,7 +1105,7 @@ def rnaseq(
             name: LatchBio
             email:
             github:
-        repository: github.com/latch-verified/bulk-rnaseq
+        repository: github.com/latchbio/rnaseq
```
Comment (ayushkamat) [inline wf/__init__.py]: why this change

File: wf/__init__.py [hunk 5]
```diff
@@ -695,11 +774,169 @@ def remote(suffix: str):
...
+    )
+
+
 class AlignmentTools(Enum):
     star_salmon = "Traditional Alignment + Quantification"
     salmon = "Selective Alignment + Quantification"
 
 
+@small_task
+def noop() -> None:
```
Comment (ayushkamat) [inline wf/__init__.py]: doesnt seem like this is being used anywhere

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: wf/__init__.py (hunk 3 above)
Comment (kennyworkman) [reply wf/__init__.py]: yeah this is artifact of the previous author...

**Comment by kennyworkman** [comment]
will take the opportunity to close https://github.com/latchbio/lgenome/issues/3

**Comment by kennyworkman** [comment]
as well as #9 (nm, new pr)

**Pull request merged**